cea/technologies/thermal_network_matrix.py

```diff
@@ -0,0 +1,1537 @@
+
+    The hydraulic thermal calculation is based on a heat balance for each edge (heat at the pipe inlet equals heat at
+    the outlet minus heat losses through the pipe). Finally, the pressure loss calculation is carried out based on
+    Todini et al. (1987)
+
+    :param locator: an InputLocator instance set to the scenario to work on
+    :param gv: an instance of globalvar.GlobalVariables with the constants  to use (like `list_uses` etc.)
+    :param network_type: a string that defines whether the network is a district heating ('DH') or cooling ('DC')
+                         network
+    :param source: string that defines the type of source file for the network to be imported ('csv' or 'shapefile')
+
+    :type locator: InputLocator
+    :type gv: GlobalVariables
+    :type network_type: str
+    :type source: str
+
+    The following files are created by this script, depending on the network type defined in the inputs:
+    - DH_EdgeNode or DC_EdgeNode: .csv
+        edge-node matrix for the defined network
+    - DH_AllNodes or DC_AllNodes: .csv
+        list of plant nodes and consumer nodes and their corresponding building names
+    - DH_MassFlow or DC_MassFlow: .csv
+        mass flow rates at each edge for each time step
+    - DH_T_Supply or DC_T_Supply: .csv
+        describes the supply temperatures at each node at each type step
+    - DH_T_Return or DC_T_Return: .csv
+        describes the return temperatures at each node at each type step
+    - DH_Plant_heat_requirement or DC_Plant_heat_requirement: .csv
+        heat requirement at from the plants in a district heating or cooling network
+    - DH_P_Supply or DC_P_Supply: .csv
+        supply side pressure for each node in a district heating or cooling network at each time step
+    - DH_P_Return or DC_P_Return: .csv
+        supply side pressure for each node in a district heating or cooling network at each time step
+    - DH_P_DeltaP or DC_P_DeltaP.csv
+        pressure drop over an entire district heating or cooling network at each time step
+
+    ..[Todini & Pilati, 1987] Todini & Pilati. "A gradient method for the analysis of pipe networks," in Computer
+    Applications in Water Supply Volume 1 - Systems Analysis and Simulation, 1987.
+    ..[Oppelt, T., et al., 2016] Oppelt, T., et al. Dynamic thermo-hydraulic model of district cooling networks.
+    Applied Thermal Engineering, 2016.
+    ..[Ikonen, E., et al, 2016] Ikonen, E., et al. Examination of Operational Optimization at Kemi District Heating 
+    Network. Thermal Science. 2016, Vol. 20, No.2, pp.667-678.
+
+    """
+
+    # # prepare data for calculation
+
+    # read building names
+    total_demand = pd.read_csv(locator.get_total_demand())
+    building_names = total_demand['Name']
+
+    # calculate ground temperature
+    weather_file = locator.get_default_weather()
+    T_ambient = epwreader.epw_reader(weather_file)['drybulb_C']
+    T_ground = geothermal.calc_ground_temperature(T_ambient.values, gv)
```

Comment:
If possible, it would be good to avoid the dependency in GV. for this please modify the way ground temperature is calculated.